epicsdbbuilder/const_array.py

```diff
@@ -0,0 +1,92 @@
+from decimal import Decimal
+from .recordbase import quote_string
+from .parameter import Parameter
+
+__all__ = ['ConstArray']
+
+
+class ConstArray:
+    """Constant Link Values. EPICS Base 3.16.1 and above.
+
+    Example: PY Source
+    ------------------
+    `r = records.lsi('r', INP=ConstArray(['Plain String not DBLINK']))`
+
+    Example: Generated DB
+    ---------------------
+    `field(INP, ["Plain String not DBLINK"])`
+    """
+
+    def __init__(self, value):
+        """Constructor.
+
+        Parameters
+        ----------
+        value : iterator
+            Homogeneous non-empty list of values.
+        """
+        self.value = value
+
+    @property
+    def value(self):
+        """A list of values.
+        """
+        return self._value
```

Comment:
I am going to update to update the code as recommended. Yet, let me explain the reason.

PEP8 (https://www.python.org/dev/peps/pep-0008/#id47) states following:
* "Use one leading underscore only for non-public methods and instance variables."
* "Generally, double leading underscores should be used only to avoid name conflicts with attributes in classes designed to be subclassed."

However, I understand that consistency across the code is important. Which coding standard is used with epicsdbbuilder?

Reply:
Well, there's something special about `__` variables: they're pseudo private, which means that users of the class can't (readily) access them.  If you use `__value` and a user tries to write, eg `ConstArray([x]).__value` they'll get an error message.  If we're going to validate on assignment to the `ConstArray` then we don't want users going being our backs and doing their own assignment to `_value`.